core/src/main/java/com/microsoft/applicationinsights/internal/agent/CoreAgentNotificationsHandler.java

```diff
@@ -0,0 +1,234 @@
+            }
+
+            String url = metaData.getURL();
+            startMethod(InstrumentedClassType.SQL, name, url, sqlStatement);
+
+        } catch (SQLException e) {
+        }
+    }
+
+    @Override
+    public String getName() {
+        return name;
+    }
+
+    @Override
+    public void onDefaultMethodEnter(String name) {
+        startMethod(InstrumentedClassType.OTHER, name, new String[]{});
+    }
+
+    @Override
+    public void onMethodFinish(String name, Throwable throwable) {
+        finalizeMethod(null, true);
+    }
+
+    @Override
+    public void onMethodFinish(String name) {
+        finalizeMethod(null, false);
+    }
+
+    private void startMethod(InstrumentedClassType type, String name, String... arguments) {
+        long start = System.nanoTime();
+
+        ThreadData localData = threadDataThreadLocal.get();
+        MethodData methodData = new MethodData();
+        methodData.interval = start;
+        methodData.type = type;
+        methodData.arguments = arguments;
+        methodData.name = name;
+        localData.methods.addFirst(methodData);
+    }
+
+    private void finalizeMethod(Object result, boolean isException) {
+        long finish = System.nanoTime();
+
+        ThreadData localData = threadDataThreadLocal.get();
+        if (localData.methods == null || localData.methods.isEmpty()) {
+            InternalLogger.INSTANCE.error("Agent has detected a 'Finish' method event without a 'Start'");
+            return;
+        }
+
+        MethodData methodData = localData.methods.removeFirst();
+        if (methodData == null) {
+            return;
+        }
+
+        methodData.interval = finish - methodData.interval;
+        methodData.result = result;
+
+        report(methodData, isException);
+    }
+
+    private void report(MethodData methodData, boolean isException) {
+
+        switch (methodData.type) {
+            case SQL:
+                sendSQLTelemetry(methodData, isException);
+                break;
+
+            case HTTP:
+                sendHTTPTelemetry(methodData, isException);
+                break;
+
+            default:
+                sendInstrumentationTelemetry(methodData, isException);
+                break;
+        }
+    }
+
+    private void sendInstrumentationTelemetry(MethodData methodData, boolean isException) {
+        Duration duration = new Duration(nanoToMilliseconds(methodData.interval));
+        RemoteDependencyTelemetry telemetry = new RemoteDependencyTelemetry(methodData.name, null, duration, !isException);
+        telemetry.setDependencyKind(DependencyKind.Undefined);
+
+        InternalLogger.INSTANCE.trace("Sending RDD event for '%s'", methodData.name);
+
+        telemetryClient.track(telemetry);
+    }
+
+    private void sendHTTPTelemetry(MethodData methodData, boolean isException) {
+        if (methodData.arguments != null && methodData.arguments.length == 1) {
+            String url = methodData.arguments[0];
```

Comment:
No argument means there is no URL data, so I think there is no much help for the user that there was an HTTP call but we don't know what the URL was, but this is a very small thing to change, I will consult Harel on that issue